### DISCUSSION TITLE
Using pretrained models

### DISCUSSION THREAD

**Discussion by gandroz** [speechbrain/speechbrain#1645] [Q&A]
I downloaded the files on HuggingFace here `https://huggingface.co/speechbrain/asr-wav2vec2-commonvoice-fr`.
However, when I try to load the model with `EncoderASR.from_hparams()`, I have the error 

```
OSError: Can't load feature extractor for 'LeBenchmark/wav2vec2-FR-7K-large'. Make sure that:

- 'LeBenchmark/wav2vec2-FR-7K-large' is a correct model identifier listed on 'https://huggingface.co/models'

- or 'LeBenchmark/wav2vec2-FR-7K-large' is the correct path to a directory containing a preprocessor_config.json file
```

My question is then, where can I define the path to the `LeBenchmark/wav2vec2-FR-7K-large` model ? I saw in the `hyperparams.yaml` file a reference to it, but apparently it is the URL on the HuggingFace Hub. How to make it run locally i.e. with not internet connection ?

**Comment by TParcollet** [comment]
Hi, I don't understand, I can't see any issue: 

<img width="957" alt="Capture d’écran 2021-09-17 à 17 49 07" src="https://user-images.githubusercontent.com/11910731/133817288-518d7f81-79a4-427a-becc-7cac33c7eb0c.png">

**Comment by gandroz** [comment]
The point is that I dont have any internel connection. I downloaded all the file in the repo mentionned above, but it tries to fetch the `LeBenchmark/wav2vec2-FR-7K-large` model from the hub too. I can surely download it too, but where do I have to save it so that speechbrain api can find it ?

**Comment by TParcollet** [comment]
aaaaaah, anywhere. Then you can simply give the path instead of the hub ID.

**Comment by gandroz** [comment]
in the `hyperparams.yaml` file ?

here:

```
# ################################
# Model: wav2vec2 + DNN + CTC/Attention
# Augmentation: SpecAugment
# Authors: Titouan Parcollet 2021
# ################################

sample_rate: 16000
wav2vec2_hub: LeBenchmark/wav2vec2-FR-7K-large     <-------HERE?

# BPE parameters
token_type: unigram  # ["unigram", "bpe", "char"]
character_coverage: 1.0

# Model parameters
activation: !name:torch.nn.LeakyReLU
dnn_layers: 2
dnn_neurons: 1024
emb_size: 128
dec_neurons: 1024
```

**Comment by TParcollet** [comment]
Yep.

**Comment by gandroz** [comment]
Merci Titouan !

**Comment by gandroz** [comment]
It is still a bit awkward to specify the model path in the method call AND in the hyperparams.yaml file...

If I write

```
model_path = Path("C:/", "Users", "ang004", "src", "stt", "model", "speechbrain-asr-wav2vec2-commonvoice-fr")
asr_model = EncoderASR.from_hparams(source=model_path, savedir=model_path)
```

I got the error mentionned above, but if I also specify 

```
wav2vec2_hub:  C:\Users\ang004\src\stt\model\speechbrain-asr-wav2vec2-commonvoice-fr
```

it works. It is really confusing to specify the model path in the `from_hparams` method and see the behavior changed by the yaml file.

**Comment by Gastron** [comment]
I'll note here that this issue and also issue #1000 both seem to stem from the fetching functionality being a little too clever. We should really consider making that simpler, and at that same time we could make some changes for the Windows issues that've arisen.

**Comment by TParcollet** [comment]
@Gastron what should we do about that then ?

**Comment by Gastron** [comment]
I don't know. Would have to find time to figure something out. It's not broken though so I wouldn't give it a high priority.
